src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersions.java

```diff
@@ -127,6 +128,54 @@ public boolean containsAnyOf(ArtifactVersions versions) {
     return artifactVersions.get().sortByReleaseDate();
   }
 
+  /**
+   * Filter {@link ArtifactVersions} by major version.
+   *
+   * @param version The artifact version.
+   * @return Filtered {@link ArtifactVersions}.
+   */
+  public ArtifactVersions filterArtifactsByMajorVersion(String version) {
+    Optional<SemanticVersion> currentVersion = SemanticVersion.parse(version);
+    if (!currentVersion.isPresent()) {
+      return ArtifactVersions.ofNothing();
+    }
+
+    String currentMajorVersion = buildMajorVersion(currentVersion.get().getMajor());
+    Optional<String> nextMajorVersion = nextMajorVersionOf(version);
```

Comment:
Here you already know that `version` is semantic, so you can guess major branches without parsing `version` again in `nextMajorVersionOf()`.

```suggestion
    int major = currentVersion.get().getMajor();
    ComparableVersion currentMajorVersion = new ComparableVersion(String.format("%s.0.0", major));
    ComparableVersion nextMajorVersion = new ComparableVersion(String.format("%s.0.0", major + 1));
```